sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_client_async.py

```diff
@@ -263,10 +267,9 @@ async def open_async(self, connection=None):
         self._network_trace_params["amqpConnection"] = self._connection._container_id
         self._network_trace_params["amqpSession"] = self._session.name
         self._shutdown = False
-        # TODO: Looks like this is broken - should re-enable later and test
-        # correct empty frame behaviour
-        # if self._keep_alive_interval:
-        #    self._keep_alive_thread = asyncio.ensure_future(self._keep_alive_async())
+
+        if self._keep_alive_interval:
+            self._keep_alive_thread = asyncio.ensure_future(self._keep_alive_async())
```

Comment:
Regarding the empty frame behaviour - have you actually seen empty frames in the amqp trace? I don't know if it's ever been tested....

Reply:
When running a keep receiver alive check, I am not seeing any empty frames coming in across the wire or going out

Reply:
I have seen empty frames come across in EH when the service wants to let the client know there aren't any messages for it to send. I would think the keep alive thread would ignore it completely and let the thread receiving it deal with it.